src/constructorio-id.js

```diff
@@ -267,6 +273,12 @@
     }
 
     if (this.session_id_storage_location === 'cookie') {
+      if (this.cookie_domain) {
+        // Delete the existing cookie set without a domain so it can be set with the same value with a domain
+        this.delete_cookie(this.cookie_name_session_id);
+        this.delete_cookie(this.cookie_name_session_data);
+      }
```

Comment:
We may be able to improve clarity here. The `if` statement reads "if there is a cookie domain" and then the comment says "delete the existing cookie without a domain", which feels confusing. I am having trouble following.